src/Microsoft.Data.Analysis/DataFrameRowCollection.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Data.Analysis
+{
+    /// <summary>
+    /// Represents the rows of a <see cref="DataFrame"/>
+    /// </summary>
+    public class DataFrameRowCollection : IEnumerable<DataFrameRow>
+    {
+        private long _currentRowIndex;
+        private DataFrame _dataFrame;
+
+        /// <summary>
+        /// Initializes a <see cref="DataFrameRowCollection"/>.
+        /// </summary>
+        internal DataFrameRowCollection(DataFrame dataFrame)
+        {
+            _dataFrame = dataFrame ?? throw new ArgumentNullException(nameof(dataFrame));
+        }
+
+        /// <summary>
+        /// An indexer to return the <see cref="DataFrameRow"/> at <paramref name="index"/>
+        /// </summary>
+        /// <param name="index">The row index</param>
+        public DataFrameRow this[long index]
+        {
+            get
+            {
+                return new DataFrameRow(_dataFrame, index);
+            }
+        }
+
+        /// <summary>
+        /// Returns an enumerator of <see cref="DataFrameRow"/> objects
+        /// </summary>
+        public IEnumerator<DataFrameRow> GetEnumerator()
+        {
+            while (_currentRowIndex < _dataFrame.RowCount)
+            {
+                yield return new DataFrameRow(_dataFrame, _currentRowIndex);
+                _currentRowIndex++;
+            }
+        }
+
+        /// <summary>
+        /// The number of rows in this <see cref="DataFrame"/>.
+        /// </summary>
+        public long Length => _dataFrame.RowCount;
```

Comment:
A couple problems with this API:

1. `Length` vs `Count`. This is a terrible .NET-ism, but array-like things use "Length", while collection like things (ICollection, etc) use "Count".
2. Why have both DataFrame.RowCount and have DataFrame.Rows.Length?  We should have just 1 way of getting the number of rows.

Reply:
Does the return type matter for `Length` vs `Count`? That's the reason I chose `Length`. `DataFrameRowCollection` is a collection, but it can have `long` number of rows, which `Count` usually returns int.

2) Hmm, I added `Length` here as an afterthought. I wanted a way for users to query the number of rows in the DataFrame from a `DataFrameRowCollection` object. To be consistent with how `Column` works, I'll keep the `Length` here and delete the `RowCount` API from `DataFrame`

Reply:
I don't think the return type matters, I think it is a "what does this class represent and how is the word used?" question.

Since this class is named a XXXCollection, I think the term `Count` makes more sense. "Length" implies that they are logically contiguous. This is the "Length" of an array, string, or a stream. But if you had a set of numbers, or a dictionary, you wouldn't use the term "Length".